Hibernate 6: Entity in Key not returned when querying

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

storage

### Describe the bug

After migrating from Hibernate 5 to Hibernate 6 in #17664, there is the exception:

```
Caused by: org.hibernate.HibernateException: identifier of an instance of org.keycloak.models.jpa.entities.FederatedIdentityEntity was altered from org.keycloak.models.jpa.entities.FederatedIdentityEntity$Key@3970c62a to org.keycloak.models.jpa.entities.FederatedIdentityEntity$Key@cf2e436d
KcOidcBrokerTest ++ 	at org.hibernate.event.internal.DefaultFlushEntityEventListener.checkId(DefaultFlushEntityEventListener.java:93)
KcOidcBrokerTest ++ 	at org.hibernate.event.internal.DefaultFlushEntityEventListener.getValues(DefaultFlushEntityEventListener.java:175)
KcOidcBrokerTest ++ 	at org.hibernate.event.internal.DefaultFlushEntityEventListener.onFlushEntity(DefaultFlushEntityEventListener.java:134)
KcOidcBrokerTest ++ 	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:127)
KcOidcBrokerTest ++ 	at org.hibernate.event.internal.AbstractFlushingEventListener.flushEntities(AbstractFlushingEventListener.java:221)
KcOidcBrokerTest ++ 	at org.hibernate.event.internal.AbstractFlushingEventListener.flushEverythingToExecutions(AbstractFlushingEventListener.java:90)
KcOidcBrokerTest ++ 	at org.hibernate.event.internal.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:38)
KcOidcBrokerTest ++ 	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:127)
KcOidcBrokerTest ++ 	at org.hibernate.internal.SessionImpl.doFlush(SessionImpl.java:1412)
```

### Version

quarkus3 branch

### Expected behavior

There shouldn't be this exception

### Actual behavior

An exception: The query returns the entity without the `user` set (it is `null`)

### How to Reproduce?

Run the test `KcOidcBrokerTest` with Quarkus

### Anything else?

_No response_

--- a/model/jpa/src/main/java/org/keycloak/models/jpa/JpaUserProvider.java
+++ b/model/jpa/src/main/java/org/keycloak/models/jpa/JpaUserProvider.java
@@ -927,16 +927,7 @@ public class JpaUserProvider implements UserProvider, UserCredentialStore {
         query.setParameter("identityProvider", identityProvider);
         query.setLockMode(lockMode);
         List<FederatedIdentityEntity> results = query.getResultList();
-        FederatedIdentityEntity federatedIdentityEntity = results.size() > 0 ? results.get(0) : null;
-        if (federatedIdentityEntity != null) {
-            // When upgrading from Hibernate 5 to Hibernate 6.2.0.CR3, the `user` wasn't filled in the returned row any more.
-            // This serves as a workaround until the cause has been found and the upstream project can provide a fix
-            // https://github.com/keycloak/keycloak/issues/19323
-            if (federatedIdentityEntity.getUser() == null) {
-                federatedIdentityEntity.setUser(userEntity);
-            }
-        }
-        return federatedIdentityEntity;
+        return results.size() > 0 ? results.get(0) : null;
     }
 
 

